Loki shooter + Coral Funnel Subsystem

Material-Energy opened the pull request.

Material-Energy committed:
Began implementing Coral Shooter and Coral Funnels

Co-authored-by: Calvin Ye <cye2007@users.noreply.github.com>
Co-authored-by: Jerry Jiang <JerryJiang531@users.noreply.github.com>
Co-authored-by: LinAndy10 <LinAndy10@users.noreply.github.com>
Co-authored-by: Weifen Chen <WennieC@users.noreply.github.com>
Co-authored-by: Rahul Deb <rahuldeb.usa@gmail.com>

Rahuldeb5 committed:
Added stall detection to shooter and implemented motor for funnel

Rahuldeb5 committed:
Added default funnel command

Material-Energy committed:
reworked funnel default command

only calls set rpm at the end

WennieC committed:
Added funnel ramp motor voltage

Material-Energy committed:
Added the shoot coral command
Moved the shooter rpm commands to the abstract class

WennieC committed:
Finished Shooter commands, reworked algae detection, added setter for targetRPM

Co-authored-by: Calvin Ye <cye2007@users.noreply.github.com>
Co-authored-by: sixuanl <sixuanl@users.noreply.github.com>
Co-authored-by: ryanyu32 <ryanyu32@users.noreply.github.com>
Co-authored-by: metereel <Material-Energy@users.noreply.github.com>
Co-authored-by: LinAndy10 <LinAndy10@users.noreply.github.com>

WennieC committed:
changed ramp rate constants thingy uwu

Material-Energy committed:
Fixed code review

Co-authored-by: Weifen Chen <WennieC@users.noreply.github.com>
Co-authored-by: Rahul Deb <rahuldeb.usa@gmail.com>

Material-Energy committed:
Added isFinished and ened methods to shooter commands

Material-Energy committed:
Made all instant command normal commands

Readded stopping functionality back to funnel default command

renamed changeState to runState in default command for clarity

removed this in all methods

changed detectCurrentSpike to private

Material-Energy committed:
Changed the lokishooter in funnel default command to an instance variable

Material-Energy committed:
Added ramp rate to shooter

k4limul commented on src/main/java/com/stuypulse/robot/commands/lokishooter/ShooterShootFront.java:
```diff
@@ -0,0 +1,34 @@
+    
+    @Override
+    public void initialize(){
+        if (shooter.hasAlgae()){
```
remove algae scoring front/back

k4limul commented on src/main/java/com/stuypulse/robot/constants/Ports.java:
```diff
@@ -12,4 +12,16 @@ public interface Gamepad {
+
+    // Set values later
+    public interface Shooter {
+        int MOTOR = 0;
```
shooter = rename to receiver port
funnel = rename to IR

k4limul commented on src/main/java/com/stuypulse/robot/constants/Settings.java:
```diff
@@ -14,4 +13,35 @@
-public interface Settings {}
+public interface Settings {  
+    public interface Shooter {
+        SmartNumber ALGAE_FRONT_SPEED = new SmartNumber("Algae Target Speed", 1);
```
no algae front/back needed

k4limul commented on src/main/java/com/stuypulse/robot/constants/Settings.java:
```diff
@@ -14,4 +13,35 @@
+
+    public interface Funnel {
+        SmartNumber MOTOR_SPEED = new SmartNumber("Funnel Speed", 0.0);
+        double BB_DEBOUNCE = 0.0;
```
rename to ir debounce

k4limul reviewed (changes requested):
look at comments

Material-Energy committed:
Made Bstreams final

Changed motorBeam on shooter to receiver

Changed motorBeam on funnel to ir

Removed front and back shooting for algae, changed to just shooting

Fixed bug where coral acquired in the wrong direction

k4limul committed:
Merge branch 'main' into se/coral-shooter-funnel

k4limul commented:
GG

k4limul committed:
add curly bracket

k4limul committed:
Rename ShooterShootAlgae.java
